console/src/components/Common/TableCommon/foldableTable/index.tsx

```diff
@@ -0,0 +1,162 @@
+import React, { ComponentProps, useCallback, useMemo } from 'react';
+import ReactTable from 'react-table';
+
+import styles from '../Table.scss';
+import { exists } from '../../utils/jsUtils';
+
+type FoldIconProps = {
+  collapsed: boolean;
+  name: string;
+};
+const FoldIcon: React.FC<FoldIconProps> = ({ collapsed, name }) => {
+  if (collapsed) {
+    return (
+      <i
+        className="fa fa-caret-right"
+        title={name ? `Expand column "${name}"` : 'Expand column'}
+      />
+    );
+  }
+
+  return <i className="fa fa-caret-left" title="Collapse column" />;
+};
+
+type FoldableHeaderProps = {
+  column: Column;
+  isFolded: boolean;
+  onFold(c: Column): void;
+};
+const FoldableHeader: React.FC<FoldableHeaderProps> = ({
+  column,
+  isFolded,
+  onFold,
+}) => {
+  const onClick = (e: React.MouseEvent<HTMLDivElement>) => {
```

Comment:
I think `useCallback` makes this better

Reply:
This is small component and `useCallback` brings more complexity. There are some cases when `useCallback` may be useful (many props, the function uses one or two), but otherwise, it's redundant.